### PR TITLE
[EPPETA-24] Refresh Token Flow

### PR DESCRIPTION
**Pull request by jleiva-gap**
Add refresh token to the authorization controller

### CODE REVIEW

File: src/webapi/Controllers/AuthorizationController.cs [hunk 1]
```diff
@@ -39,36 +41,77 @@ public class AuthorizationController : Controller
...
     public async Task<IActionResult> Exchange([FromForm] PasswordTokenRequest tokenRequest)
     {
-        if (tokenRequest.GrantType != GrantTypes.Password)
+        var openIdRequest = HttpContext.GetOpenIddictServerRequest();
+        if (openIdRequest?.GrantType != GrantTypes.Password && openIdRequest?.GrantType != GrantTypes.RefreshToken)
         {
             return BadRequest(new { error = "The application only accepts the password grant type at this time." });
         }
 
         var user = await _userManager.FindByNameAsync(tokenRequest.Username);
```
Comment (stevenarnold-ed-fi) [inline src/webapi/Controllers/AuthorizationController.cs]: if this is a refresh token call, we immediately null the user out. Can we move this call under the `else if (openIdRequest.GrantType == GrantTypes.Password)` call on line 88

File: src/webapi/Controllers/AuthorizationController.cs [hunk 2]
```diff
@@ -82,18 +125,18 @@ public async Task<IActionResult> Exchange([FromForm] PasswordTokenRequest tokenR
                 .SetClaim(Claims.Email, await _userManager.GetEmailAsync(user))
                 .SetClaim(Claims.Name, $"{user.FirstName} {user.LastName}")
                 .SetClaims(Claims.Role, (await _userManager.GetRolesAsync(user)).ToImmutableArray());
-
+        tokenRequest.Scopes = "offline_access";
```
Comment (stevenarnold-ed-fi) [inline src/webapi/Controllers/AuthorizationController.cs:129]: What does this do? We now have a ` "scope":"offiline_access" ` in every token. Was that desired?

**Review by stevenarnold-ed-fi — CHANGES_REQUESTED** [review]

File: src/webapi/Controllers/AuthorizationController.cs (hunk 1 above)
Comment (jleiva-gap) [reply src/webapi/Controllers/AuthorizationController.cs]: Done!

File: src/webapi/Controllers/AuthorizationController.cs (hunk 2 above)
Comment (jleiva-gap) [reply src/webapi/Controllers/AuthorizationController.cs:129]: I added that scope to return the refresh token. I attach the documentation that I read to implement this: **[The offline access scope](https://learn.microsoft.com/en-us/entra/identity-platform/scopes-oidc#the-offline_access-scope)**

**Review by stevenarnold-ed-fi — APPROVED** [review]

**Review by stevenarnold-ed-fi — APPROVED** [review]

**Pull request merged**